File: homework1/JAVA/Main.java
package JAVA;

import java.util.List;

public class Main {

    public static void main(String[] args) {

        List<Card> cards = List.of(
            () -> true,
            () -> false,
            () -> true,
            () -> false
        );

        CardOrganizer sorter = new RecentMistakesFirstSorter();
        List<Card> sorted = sorter.organize(cards);

        for (Card c : sorted) {
            System.out.println(c.wasCorrectLastRound());
        }
    }
}

File: homework1/JS/main.ts
import { newRecentMistakesFirstSorter } from "./newRecentMistakesFirstSorter";

const cards = [
    { wasCorrectLastRound: () => true },
    { wasCorrectLastRound: () => false },
    { wasCorrectLastRound: () => true },
    { wasCorrectLastRound: () => false }
];

const sorter = new newRecentMistakesFirstSorter();
const result = sorter.organize(cards);

console.log(result.map(c => c.wasCorrectLastRound()));
